src/components/LoadingSpinner.tsx

```diff
@@ -64,20 +80,43 @@ const LoadSpinnerIconContainer = styled.div<{ $fontSize: number }>`
  */
 export default function LoadingSpinner(inputProps: PropsWithChildren<LoadingSpinnerProps>): ReactElement {
   const props = { ...defaultProps, ...inputProps } as PropsWithChildren<Required<LoadingSpinnerProps>>;
+
+  // Delay showing progress bar slightly; this fixes a visual bug where the loading spinner
+  // would flash the progress bar at 100% right as it vanished.
+  const showProgressBar = useDebounce(props.progress !== undefined && props.progress !== null, VANISH_DURATION_MS);
+
+  // Disable completion checkmark by forcing value to always be shown as a number %
+  const progressFormatter = (percent?: number): ReactNode => {
+    if (percent === undefined) {
+      return "";
+    }
+    return `${percent}%`;
+  };
```

Comment:
nit: doesn't necessarily need to be defined within the component body

Reply:
Fixed!